controllers/backup.go

```diff
@@ -78,55 +82,333 @@ func (r *BackupReconciler) getActiveBackupName(backup *v1beta1.Backup, c client.
+		// skip if unknown backup type
+		return
+	}
+
+	backupLogger := log.FromContext(ctx)
+	// create resources backup
+	veleroResIdentity := types.NamespacedName{
+		Namespace: veleroBackup.Namespace,
+		Name:      fmt.Sprintf("%s-%s", veleroBackup.Name, resourceType),
+	}
+
+	veleroResBackup := &veleroapi.Backup{}
+	veleroResBackup.Name = veleroResIdentity.Name
+	veleroResBackup.Namespace = veleroResIdentity.Namespace
+
+	errBackup := r.Get(ctx, veleroResIdentity, veleroResBackup)
+	if errBackup != nil {
+		backupLogger.Info("velero.io.Backup [name=%s, namespace=%s] returned error, checking if the resource was not yet created", veleroResIdentity.Name, veleroResIdentity.Namespace)
+		// check if this is a  resource NotFound error, in which case create the resource
+		if k8serr.IsNotFound(errBackup) {
+			// create backup based on resource type
+			switch resourceType {
+			case "resource":
+				setResourcesBackupInfo(ctx, veleroResBackup, c)
+			case "creds":
+				setCredsBackupInfo(ctx, veleroResBackup, c)
+			}
+
+			errBackup = c.Create(ctx, veleroResBackup, &client.CreateOptions{})
+			if veleroResBackup != nil {
+				// add backup to list of backups
+				backup.Status.VeleroBackups = append(backup.Status.VeleroBackups, veleroResBackup)
+			}
+
+			if errBackup != nil {
+				backupLogger.Error(errBackup, "create backup error")
+				return
+			}
+		}
+	} else {
+		if resourceType == "creds" {
+			backup.Status.VeleroBackups[1] = veleroResBackup
+		} else {
+			backup.Status.VeleroBackups[2] = veleroResBackup
+
+		}
+	}
+
+	if veleroResBackup.Status.Progress != nil {
+		msg := fmt.Sprintf("%s ; [%s: ItemsBackedUp[%d], TotalItems[%d]]", backup.Status.LastMessage, resourceType, veleroResBackup.Status.Progress.ItemsBackedUp, veleroResBackup.Status.Progress.TotalItems)
+		backup.Status.LastMessage = msg
+	}
 }
 
-// return Duration in format 1h15m30s
-func getFormattedDuration(duration time.Duration) string {
+// set all acm resources backup info
+func setResourcesBackupInfo(ctx context.Context, veleroBackup *veleroapi.Backup, c client.Client) {
+
+	backupLogger := log.FromContext(ctx)
+	var clusterResource bool = false
+	veleroBackup.Spec.IncludeClusterResources = &clusterResource
+	veleroBackup.Spec.ExcludedNamespaces = appendUnique(veleroBackup.Spec.ExcludedNamespaces, "local-cluster")
+
+	for i := range backupResources { // acm resources
+		veleroBackup.Spec.IncludedResources = appendUnique(veleroBackup.Spec.IncludedResources, backupResources[i])
+	}
 
-	formatted := duration.Truncate(time.Second).String()
-	return formatted
+	// exclude acm channel namespaces
+	channels := chnv1.ChannelList{}
+	if err := c.List(ctx, &channels, &client.ListOptions{}); err != nil {
+		// if NotFound error
+		if !k8serr.IsNotFound(err) {
+			backupLogger.Info("channel resources NOT FOUND")
+		} else {
+			backupLogger.Error(err, "failed to get chnv1.ChannelList")
+		}
+	} else {
+		for i := range channels.Items {
+			if channels.Items[i].Name == "charts-v1" {
```

Comment:
I need help here. Is `charts-v1` something that will never change or be configurable externally?

Reply:
yes, we expect this does not change; rather then assume the acm install ns is open-cluster-management, I can use this channel name, used by installer to find the ns name
We've used this channel channel name with the app as well